app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
+                        "links": topology_update["links"],
+                        }
+            else:
+                self.sdx_topology = topology_mock.topology_mock()
+            evaluate_topology = self.validate_sdx_topology()
+            if evaluate_topology["status_code"] == 200:
+                result = self.post_sdx_lc(event_type)
+                return result
+            return {"result": evaluate_topology['result'],
+                    "status_code": evaluate_topology['status_code']}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
+        return {"result": "No SDX Topology loaded", "status_code": 401}
+
     @listen_to("kytos/topology.*")
-    def listen_topology(self, event=None):
+    def listen_event(self, event=None):
         """Function meant for listen topology"""
-        f_name = " listen_topology "
-        log.info(f"{HSH}{f_name}get kytos topology {HSH}")
-        event_info = {}  # pylint: disable=W0201
-        if event:
-            allowed_events = [
-                "kytos/topology.switch.enabled",
-                "kytos/topology.switch.disabled",
-                "kytos/topology.link_up",
-                "kytos/topology.link_down",
-            ]
-            if event.name in allowed_events:
-                try:
-                    topology = requests.get(settings.KYTOS_TOPOLOGY_URL).json()
-                    log.info(
-                        f"{HSH}{f_name}constructor post request {HSH}"
-                    )
-                    event_info = {
-                        "event": event,
-                        "kytostopology": topology,
-                    }  # pylint: disable=W0201
-                    try:
-                        requests.post(settings.SDX_CONSTRUCTOR).json(event_info)
-                    except Exception as err:  # pylint: disable=W0703
-                        log.info(f"{HSH}{f_name}constructor post error {HSH}")
-                        log.info(err, event_info)
-                except Exception as err:  # pylint: disable=W0703
-                    log.info(
-                            f"{HSH}{f_name}get kytos topology error {HSH}"
-                    )
-                    log.info(err)
-        return event_info
-
-    @rest("v2/listen_topology", methods=["GET"])
-    def get_listen_topology(self):
-        """consume call listen Topology"""
-        f_name = " get_listen_topology "
+        f_name = " listen_event "
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        log.info(f"{HSH}{f_name} {HSH}")
+        log.info(f"{HSH} dir(event): {dir(event)} {HSH}")
+        log.info(f"{HSH} event: {event} {HSH}")
+        log.info(f"{HSH} event.name: {event.name} {HSH}")
+        log.info(f"{HSH} event.content: {event.content} {HSH}")
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        if event is not None and self.get_kytos_topology():
+            if event.name in settings.ADMIN_EVENTS:
+                event_type = 1
```

Comment:
It'd be easier to follow if the `event_type` were a string carrying the semantics of what it represents. I found myself searching for this value again, when trying to check what the event_type integer value meant.

Reply:
I agree that the event_type does not indicate a mathematical operation. Rather, it means an administrative event that denotes the execution of an action by an operator that leads to a topology change event, or an operational event that arises from the network itself.

Reply:
I will modify it to enhance its descriptive quality by replacing "1 or 2" with "operational or administrative." Thank you.

Reply:
Looks good